common/src/main/java/org/opensearch/ml/common/MLConfig.java

```diff
@@ -74,23 +106,42 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         out.writeOptionalInstant(createTime);
         out.writeOptionalInstant(lastUpdateTime);
+        if (streamOutputVersion.onOrAfter(MINIMAL_SUPPORTED_VERSION_FOR_NEW_CONFIG_FIELDS)) {
+            out.writeOptionalString(configType);
+            if (mlConfiguration != null) {
+                out.writeBoolean(true);
+                mlConfiguration.writeTo(out);
+            } else {
+                out.writeBoolean(false);
+            }
+            out.writeOptionalInstant(lastUpdatedTime);
+        }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder xContentBuilder, Params params) throws IOException {
         XContentBuilder builder = xContentBuilder.startObject();
         if (type != null) {
-            builder.field(CONFIG_TYPE_FIELD, type);
+            builder.field(TYPE_FIELD, type);
+        }
+        if (configType != null) {
+            builder.field(CONFIG_TYPE_FIELD, configType);
         }
         if (configuration != null) {
-            builder.field(ML_CONFIGURATION_FIELD, configuration);
+            builder.field(CONFIGURATION_FIELD, configuration);
+        }
+        if (mlConfiguration != null) {
+            builder.field(ML_CONFIGURATION_FIELD, mlConfiguration);
```

Comment:
What happens when `ml_configuration` and `configuration` are both populated. Same with `config_type` and `config`? Will this cause any issues in the plugin?

Reply:
No there wouldn't be any issue but both will be shown in the response.